servers/audio/audio_stream.cpp

```diff
@@ -351,24 +350,36 @@ int AudioStreamPlaybackMicrophone::_mix_internal(AudioFrame *p_buffer, int p_fra
 					input_ofs = 0;
 				}
 
-				p_buffer[i] = AudioFrame(l, r);
+				p_buffer[current_frame] = AudioFrame(l, r);
 			} else {
-				if (mixed_frames == p_frames) {
-					mixed_frames = i;
+				if (!was_locked) {
+					// Wait while the other thread writes to the input buffer.
+					AudioDriver::get_singleton()->input_lock();
+					AudioDriver::get_singleton()->input_unlock();
+
+					size_position = AudioDriver::get_singleton()->get_input_size_position();
+					was_locked = true;
+
+					// Write to the same frame on the next loop iteration.
+					current_frame--;
```

Comment:
Doesn't seem that this code actually executes ever, maybe it should be removed and replaced with just printing an error?

Reply:
See #93154